exp/addons/controllers/clusterresourceset_controller_test.go

```diff
@@ -192,6 +194,35 @@ apiVersion: v1`,
 				UID:        testCluster.UID,
 			})
 		}, timeout).Should(BeTrue())
-	})
 
+		// Wait until ClusterResourceSetBinding is created for the Cluster
+		isBindingCreated := false
+		for isBindingCreated == true {
+			binding := &addonsv1.ClusterResourceSetBinding{}
+			clusterResourceSetBindingKey := client.ObjectKey{
+				Namespace: testCluster.Namespace,
+				Name:      testCluster.Name,
+			}
+			err := testEnv.Get(ctx, clusterResourceSetBindingKey, binding)
+			if err == nil {
+				isBindingCreated = true
+			}
+		}
+
+		By("Verifying ClusterResourceSetBinding is deleted when its cluster owner reference is deleted")
+		controllerutil.AddFinalizer(testCluster, "finalizer")
```

Comment:
When deleting the testCluster below, somehow DeletionTimestamp never becomes non-zero when it comes to clusterToClusterResourceSet(). Setting a finalizer here fixes that problem but normally it should not be needed. Still looking to understand why this is the case.

Reply:
/hold